s3/pkg/datastore/alibaba/alibaba.go

```diff
@@ -0,0 +1,349 @@
+		log.Error("### MD5 not match, calculatedMd5:", calculatedMd5, "userMd5:", userMd5)
+		return result, errors.New(" Error in calculating calculatedMd5")
+	}
+
+	result.UpdateTime = time.Now().Unix()
+	result.ObjectId = objectId
+	result.Etag = calculatedMd5
+	result.Written = size
+	log.Info("Upload to oss successfully.")
+	return result, nil
+}
+
+func (ad *OSSAdapter) Get(ctx context.Context, object *pb.Object, start int64, end int64) (io.ReadCloser, error) {
+	bucket := ad.backend.BucketName
+	log.Infof("bucket is %v\n", bucket)
+	log.Infof("object key %v    %v\n", object.ObjectKey, object.ObjectId)
+
+	objectId := object.BucketName + "/" + object.ObjectKey
+	alibabaBucket, err := ad.client.Bucket(bucket)
+	if err != nil {
+		log.Errorf("Access bucket failed:%v", err)
+		return nil, ErrGetFromBackendFailed
+	}
+	log.Println("Start Range ", start, " End Range  ", end)
+	log.Println(" objectId ", objectId, " object.ObjectKey  ", object.ObjectKey)
+
+	body, err := alibabaBucket.GetObject(objectId, oss.Range(start, end))
+	if err != nil {
+		log.Errorf("download object failed:%v", err)
+		return nil, ErrGetFromBackendFailed
+	}
+
+	log.Println(" download object ", body)
+	return body, nil
+
+}
+
+func (ad *OSSAdapter) Delete(ctx context.Context, object *pb.DeleteObjectInput) error {
+
+	bucket := ad.backend.BucketName
+	objectId := object.Bucket + "/" + object.Key
+	getbucket, err := ad.client.Bucket(bucket)
+	if err != nil {
+		log.Errorf("get bucket failed, err:%v\n", err)
+		return ErrDeleteFromBackendFailed
+	}
+	// Delete an object.
+	err = getbucket.DeleteObject(objectId)
+	if err != nil {
+		log.Errorf("Delete object failed, err:%v\n", err)
+		return ErrDeleteFromBackendFailed
+	}
+	log.Infof("delete object[OSS] succeed, objectId:%s.\n", objectId)
+	return nil
+
+}
+
+func (ad *OSSAdapter) ChangeStorageClass(ctx context.Context, object *pb.Object, newClass *string) error {
+
+	log.Infof("change storage class[OSS] of object[%s] to %s .\n", object.ObjectId, newClass)
+	bucket := ad.backend.BucketName
+	objectId := object.ObjectId
+	alibabaBucket, err := ad.client.Bucket(bucket)
+	srcObjectKey := object.BucketName + "/" + object.ObjectKey
+	var StorClass oss.StorageClassType
+	switch *newClass {
+	case "STANDARD_IA":
+		StorClass = oss.StorageIA
+	case "GLACIER":
+		StorClass = oss.StorageArchive
+	default:
+		log.Infof("[OSS] unSpport storage class:%s", newClass)
+		return ErrInvalidStorageClass
+	}
+	_, err = alibabaBucket.CopyObject(srcObjectKey, objectId, oss.ObjectStorageClass(StorClass))
+
+	if err != nil {
+		log.Errorf("[OSS] change storage class of object[%s] to %s failed: %v\n", object.ObjectId, newClass, err)
+		return ErrPutToBackendFailed
+	} else {
+		log.Infof("[OSS] change storage class of object[%s] to %s succeed.\n", object.ObjectId, newClass)
+	}
+
+	return nil
+}
+
+func (ad *OSSAdapter) Copy(ctx context.Context, stream io.Reader, target *pb.Object) (result dscommon.PutResult, err error) {
```

Comment:
This is not yet implemented? Please put NOIMPL log

Reply:
Fixed, Thanks.